src/com/esotericsoftware/kryo/serializers/CompatibleFieldSerializer.java

```diff
@@ -58,6 +61,22 @@ public CompatibleFieldSerializer (Kryo kryo, Class type, CompatibleFieldSerializ
 		this.config = config;
 	}
 
+	@Override
+	protected void initializeCachedFields() {
+		if (!super.config.extendedFieldNames) {

```

Comment:
Do you really need to call `super`? Why can't we use the `config` field from this class?

Reply:
config.extendedFieldNames report:
java.lang.NullPointerException at com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer.initializeCachedFields(CompatibleFieldSerializer.java:66)
	at com.esotericsoftware.kryo.serializers.CachedFields.rebuild(CachedFields.java:111)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.<init>(FieldSerializer.java:82)
	at com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer.<init>(CompatibleFieldSerializer.java:60)
	at com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer.<init>(CompatibleFieldSerializer.java:56)
	at com.esotericsoftware.kryo.serializers.CompatibleFieldSerializerTest.testLogWarningOnDuplicateFieldInClassHierarchy(CompatibleFieldSerializerTest.java:666)

Unless setExtendedFieldNames is override in CompatibleFieldSerializerConfig.
If you have a better way, please teach me . Thank you in advance.

Reply:
@theigl You are right! Done.